tlatools/org.lamport.tlatools/src/util/SimpleFilenameToStream.java

```diff
@@ -65,66 +70,43 @@ public SimpleFilenameToStream(final String libraryPath) {
  * (which is being used in the default constructor).
  */
   public SimpleFilenameToStream(String[] anLibraryPaths) {
-	  libraryPaths = getLibraryPaths(getInstallationBasePath(), anLibraryPaths);
+    userFileResolver = getLibraryPaths(anLibraryPaths);
+    standardLibraryResolver = getInstallationBasePath();
   }
 
   /**
-   * Find the absolute path in the file system to the directory
-   * containing the TLA+ standard modules; this path
-   * must have separators appropriate to the Unix ('/') or Windows ('\') world.
+   * Find the TLA+ standard modules.
    *
-   * @return the location of the TLA+ standard modules as an absolute path or as a "jar:" URL
+   * @return a resolver for locating the TLA+ standard modules
    */
-  private String getInstallationBasePath() {
-
-    // get a "file:" URL for the base directory for package tla2sany
-    final URL         url = cl.getResource(STANDARD_MODULES);
-
-    // jar expanded to the fs (make sure to handle whitespaces correctly
-    final String path = url.toString();
-	try {
-    	// convert to URI which handles paths correctly (even OS dependently)
-    	if(!FilenameToStream.isInJar(path)) {
-    	final URI uri = new URI(path);
-    		return new File(uri).getAbsolutePath();
-    	}
-    } catch (URISyntaxException e) {
-    	System.err.println(path);
-    	// may never happen
-    	e.printStackTrace();
-    } catch (IllegalArgumentException e) {
-    	System.err.println(path);
-    	// may never happen
-    	e.printStackTrace();
-    }
-    return path;
-   }
+  private static Resolver getInstallationBasePath() {
+    return new ClasspathResolver(cl, STANDARD_MODULES);
+  }
 
   /**
    * August 2014 - TL
    * added an informative method for returning the actual path used by this resolver
    */
   public String getFullPath() {
-    StringBuffer buf = new StringBuffer();
-    String[] ar = libraryPaths;
-    for (int i=0; i<ar.length; i++)
-    {
-      buf.append(ar[i]);
-      if (i <ar.length-1) {
-        buf.append(", ");
-      }
-    }
-    return buf.toString();
+    return userFileResolver.describeSearchLocations() + ", " + standardLibraryResolver.describeSearchLocations();
   }
 
   /**
    * August 2014 - TL
    * added functionality for supplying additional libraries.
    * All usage of this except the one added by TL is supplying libraries  = null
    */
-  private String[] getLibraryPaths(final String installationBasePath, String[] libraries) {
-    String[] res;
-    String path = null;
+  private static Resolver getLibraryPaths(String[] libraries) {
+    List<Resolver> res = new ArrayList<>();
+
+    // First, search the current directory
```

Comment:
Does the current behavior search the CWD or the directory that the root spec is in? I recall I opened an issue about this leading to some confusing SANY/TLC behavior a long time ago... #545

Reply:
Thanks a ton for pointing this out. See my reply on [your other comment about this](https://github.com/tlaplus/tlaplus/pull/1116#discussion_r1911428941).